nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -0,0 +1,300 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+var deploymentTypeValues = []string{
+	model.EdgeCluster_DEPLOYMENT_TYPE_VIRTUAL_MACHINE,
+	model.EdgeCluster_DEPLOYMENT_TYPE_PHYSICAL_MACHINE,
+	model.EdgeCluster_DEPLOYMENT_TYPE_UNKNOWN,
+}
+
+var memberNodeTypeValues = []string{
+	model.EdgeCluster_MEMBER_NODE_TYPE_EDGE_NODE,
+	model.EdgeCluster_MEMBER_NODE_TYPE_PUBLIC_CLOUD_GATEWAY_NODE,
+	model.EdgeCluster_MEMBER_NODE_TYPE_UNKNOWN,
+}
+
+func resourceNsxtEdgeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtEdgeClusterCreate,
+		Read:   resourceNsxtEdgeClusterRead,
+		Update: resourceNsxtEdgeClusterUpdate,
+		Delete: resourceNsxtEdgeClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
```

Comment:
Same as above - I think we should follow the convention and make this Required, and use `getDisplayNameSchema` here